packages/core/src/components/ChoiceList/Choice.jsx

```diff
@@ -11,41 +11,88 @@ import uniqueId from 'lodash.uniqueid';
  * to the `input` element, so you can use this to set additional attributes if
  * necessary.
  */
-export const Choice = function(props) {
-  /* eslint-disable prefer-const */
-  let {
-    // Using let rather than const since we sometimes rewrite id
-    children,
-    className,
-    id,
-    inversed,
-    inputPlacement,
-    inputClassName,
-    size,
-    requirementLabel,
-    ...inputProps
-  } = props;
-  /* eslint-enable prefer-const */
+export class Choice extends React.PureComponent {
+  constructor(props) {
+    super(props);
 
-  const inputClasses = classNames(inputClassName, 'ds-c-choice', {
-    'ds-c-choice--inverse': inversed,
-    'ds-c-choice--right': inputPlacement === 'right',
-    'ds-c-choice--small': size === 'small'
-  });
+    this.handleChange = this.handleChange.bind(this);
 
-  if (!id) {
-    id = uniqueId(`${inputProps.type}_${inputProps.name}_`);
+    if (typeof this.props.checked === 'undefined') {
+      this.isControlled = false;
+      // Since this isn't a controlled component, we need a way
+      // to track when the value has changed. This can then be used
+      // to identify when to toggle the visibility of (un)checkedChildren
+      this.state = { checked: this.props.defaultChecked };
+    } else {
+      this.isControlled = true;
+    }
   }
 
-  return (
-    <div className={className}>
-      <input className={inputClasses} id={id} {...inputProps} />
-      <FormLabel fieldId={id} requirementLabel={requirementLabel}>
-        {children}
-      </FormLabel>
-    </div>
-  );
-};
+  checked() {
+    if (this.isControlled) {
+      return this.props.checked;
+    }
+
+    return this.state.checked;
+  }
+
+  handleChange(evt) {
+    if (this.props.onChange) {
+      this.props.onChange(evt);
+    }
+
+    if (!this.isControlled) {
+      this.setState({ checked: evt.target.checked });
+    }
+  }
+
+  render() {
+    /* eslint-disable prefer-const */
+    let {
```

Comment:
It seems a shame for all of this to be declared with `let` when only `id` is changed.